### PR TITLE
* MDF [conf] enable bridge retain flag in old conf

### PR DESCRIPTION
**Pull request by OdyWayne**

### CODE REVIEW

File: src/supplemental/nanolib/conf.c [hunk 1]
```diff
@@ -2532,18 +2552,30 @@ conf_bridge_node_parse_forwards(
 			goto check;
 		}
 
+		snprintf(key, 128,
+		    "%s%s.forwards.%ld.retain", prefix, name,
+		    fwd_index);
+		if (!get_retain &&
+		    (value = get_conf_value(line, sz, key)) != NULL) {
+			retain = (uint8_t) atoi(value);
```
Comment (wanghaEMQ) [inline src/supplemental/nanolib/conf.c:2560]: Donot forget to limit it to 0\1\2

**Review by wanghaEMQ — APPROVED** [review]

**Review by wanghaEMQ — APPROVED** [review]

File: src/supplemental/nanolib/conf.c (hunk 1 above)
Comment (OdyWayne) [reply src/supplemental/nanolib/conf.c:2563]: ok

**Review by RanMaoyi — APPROVED** [review]

**Review by lee-emqx — APPROVED** [review]

File: src/supplemental/nanolib/test_conf/nmq_test.conf [hunk 2]
```diff
@@ -1063,3 +1138,64 @@ rules.mysql.mysql_rule_db {
...
+# 	# # The prefix of parquet files written.
+# 	# #
+# 	# # Value: string
+# 	file_name_prefix = ""
+# 	# # Maximum rotation count of parquet files.
+# 	# #
+# 	# # Value: Number
+# 	# # Default: 5
+# 	file_count = 5
+# }
```
Comment (OdyWayne) [inline src/supplemental/nanolib/test_conf/nmq_test.conf:1201]: There are plenty of memleaks in exchange & parquet now.

File: src/supplemental/nanolib/conf_ver2.c [hunk 3]
```diff
@@ -688,13 +692,15 @@ conf_auth_http_req_parse_ver2(conf_auth_http_req *config, cJSON *jso)
...
+			config->params = realloc(
+			    config->params, cnt * sizeof(conf_http_param *));
+			config->params[cnt - 1] = param;
 		} else {
 			nng_strfree(param->name);
 			NNI_FREE_STRUCT(param);
 		}
 	}
-	config->param_count = cvector_size(config->params);
+	config->param_count = cnt;
```
Comment (JaylinYu) [inline src/supplemental/nanolib/conf_ver2.c:703]: ???

File: src/supplemental/nanolib/test_conf/nmq_old_test.conf [hunk 4]
```diff
@@ -623,13 +623,13 @@ bridge.mqtt.emqx.quic_congestion_control=cubic
...
 ## Default: false
-bridge.mqtt.emqx.hybrid_bridging=true
+bridge.mqtt.emqx.hybrid_bridging=false
 
 ## multi-stream: enable or disable the multi-stream bridging mode
 ## Still a feature WIP, but basic func is ready
 ## Value: true/false
 ## Default: false
-bridge.mqtt.emqx.quic_multi_stream=true
+bridge.mqtt.emqx.quic_multi_stream=false
```
Comment (JaylinYu) [inline src/supplemental/nanolib/test_conf/nmq_old_test.conf:632]: why?

File: src/supplemental/nanolib/test_conf/nmq_test.conf (hunk 2 above)
Comment (JaylinYu) [reply src/supplemental/nanolib/test_conf/nmq_test.conf:1201]: worth an issue

**Review by JaylinYu — APPROVED** [review]
not sure the reasons of some changes.

**Review by RanMaoyi — APPROVED** [review]

File: src/supplemental/nanolib/conf_test.c [hunk 5]
```diff
@@ -105,6 +105,6 @@ NUTS_TESTS = {
    {"get size", test_get_size},
    {"get time", test_get_time},
    {"conf parse v2", test_conf_parse_ver2},
-   {"conf parse", test_conf_parse},
+//    {"conf parse", test_conf_parse},
```
Comment (JaylinYu) [inline src/supplemental/nanolib/conf_test.c:108]: remember to fire an issue regarding this

**Review by JaylinYu — APPROVED** [review]

File: src/supplemental/nanolib/conf_test.c (hunk 5 above)
Comment (OdyWayne) [reply src/supplemental/nanolib/conf_test.c:108]: roger

File: src/supplemental/nanolib/test_conf/nmq_test.conf (hunk 2 above)
Comment (wanghaEMQ) [reply src/supplemental/nanolib/test_conf/nmq_test.conf:1211]: > There are plenty of memleaks in conf parse in exchange & parquet now.

I have made some updates. Those memory leakage will no longer exists after that pr was merged.

**Pull request merged**